internal/common/function/function_test.go

```diff
@@ -409,23 +419,6 @@ func TestValidateBackupSpecEnforcedFields(t *testing.T) {
 			}
 		})
 	}
-
```

Comment:
I do not like this change

idea of this is to help us test if new fields are added to velero API

I would `t.Run` inside `for` to run different test for restricted fields

Example 
```go
            if slices.Contains([]string{"ExcludedNamespaces"}, test.name) {
				reflect.ValueOf(userNonAdminBackup.Spec.BackupSpec).Elem().FieldByName(test.name).Set(reflect.ValueOf(test.overrideValue))
				err := ValidateBackupSpec(context.Background(), fakeClient, "oadp-namespace", userNonAdminBackup, enforcedSpec)
				if err == nil {
					t.Errorf("setting restricted backup spec field '%v' test failed: %v", test.name, err)
				}
				return
			}
```

Reply:
We are not referring the actual Velero API in the tests here. So in a case when Velero API get updated our unit tests won't automatically start failing, right ? We will still need to add tests for manually testing spec values that are enforceable.  Please correct me if I am wrong here.
In the current PR we are testing values that are enforceable in `TestValidateBackupSpecEnforcedFields` and those which have validation error are tested under `TestValidateBackupSpec`. I believe this much is adequate here.

Reply:
Done, updated the PR !